src/test/ManagedDripsHub.t.sol

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.7;
+
+import {DripsHubUser, ManagedDripsHubUser} from "./DripsHubUser.t.sol";
+import {DripsHubTest} from "./DripsHub.t.sol";
+import {DripsReceiver, SplitsReceiver} from "../DripsHub.sol";
+import {ManagedDripsHub, ManagedDripsHubProxy} from "../ManagedDripsHub.sol";
+
+abstract contract ManagedDripsHubTest is DripsHubTest {
+    ManagedDripsHub private dripsHub;
+    ManagedDripsHubUser private admin;
+    ManagedDripsHubUser private user;
+    ManagedDripsHubUser private user1;
+    ManagedDripsHubUser private user2;
+
+    string private constant ERROR_NOT_ADMIN = "Caller is not the admin";
+    string private constant ERROR_PAUSED = "Contract paused";
+
+    // Must be called once from child contract `setUp`
+    function setUp(ManagedDripsHub dripsHub_) internal {
+        dripsHub = dripsHub_;
+        admin = createManagedUser();
+        dripsHub.changeAdmin(address(admin));
+        user = createManagedUser();
+        user1 = createManagedUser();
+        user2 = createManagedUser();
+        super.setUp(dripsHub);
+    }
+
+    function createUser() internal override returns (DripsHubUser) {
+        return createManagedUser();
+    }
+
+    function createManagedUser() internal virtual returns (ManagedDripsHubUser);
+
+    function wrapInProxy(ManagedDripsHub hubLogic) internal returns (ManagedDripsHub) {
+        ManagedDripsHubProxy proxy = new ManagedDripsHubProxy(hubLogic, address(this));
+        return ManagedDripsHub(address(proxy));
+    }
+
+    function testAdminCanBeChanged() public {
+        assertEq(admin.admin(), address(admin));
+        admin.changeAdmin(address(user));
+        assertEq(admin.admin(), address(user));
+    }
+
+    function testOnlyAdminCanChangeAdmin() public {
+        try user1.changeAdmin(address(user2)) {
+            assertTrue(false, "ChangeAdmin hasn't reverted");
+        } catch Error(string memory reason) {
+            assertEq(reason, ERROR_NOT_ADMIN, "Invalid changeAdmin revert reason");
+        }
+    }
+
+    function testOnlyAdminCanUpgradeContract() public {
+        try user.upgradeTo(address(0)) {
+            assertTrue(false, "ChangeAdmin hasn't reverted");
+        } catch Error(string memory reason) {
+            assertEq(reason, ERROR_NOT_ADMIN, "Invalid changeAdmin revert reason");
+        }
+        // Fails for admin too, but due to a different reason.
+        // As of now we can't reasonably test contract upgrade.
```

Comment:
Why?

Reply:
Because we only have 1 implementation of the pool, so it'd be hard to see any changes. OTOH we could make a "dumb" upgrade, which would change some constants, like cycle length :thinking:

Reply:
Exactly, something like that or even the exact same contract but just redeployed

Reply:
Fixed